### PR TITLE
Replace cdk-ec2-key-pair with CfnKeyPair construct to generate ec2 key pair

### PR DESCRIPTION
**Pull request by rishabh6788**
### Description
Replace cdk-ec2-key-pair with CfnKeyPair construct to generate ec2 key pair. 
`aws-cdk-lib` provides `CfnKeyPair` construct to generate ec2 key-pairs. The private key is saved to AWS Systems Manager Parameter Store, using a parameter with the following name: /ec2/keypair/{key_pair_id}. See https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_ec2.CfnKeyPair.html for more details. 

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._ 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -53,46 +53,48 @@ OpenSearch Continuous Integration is an open source CI system for OpenSearch and
 
 ### Dev Deployment 
 1. Setup your local machine to credentials to deploy to the AWS Account
-1. Deploy the bootstrap stack by running the following command that sets up required resources to create the stacks. [More info](https://docs.aws.amazon.com/cdk/latest/guide/bootstrapping.html)
+2. Deploy the bootstrap stack by running the following command that sets up required resources to create the stacks. [More info](https://docs.aws.amazon.com/cdk/latest/guide/bootstrapping.html)
```
Comment (peterzhuamazon) [inline README.md:56]: You can keep these 1/1/1 as they will render as 1/2/3 😄

**Review by peterzhuamazon — APPROVED** [review]

File: lib/compute/agent-node-config.ts [hunk 2]
```diff
@@ -48,12 +49,24 @@ export class AgentNodeConfig {
...
     });
-    Tags.of(key)
-      .add('jenkins:credentials:type', 'sshUserPrivateKey');
+
+    // Create a secret for storing the actaul value of key-pair so that jenkins secret-manager plugin is able to fetch it
+    const keyPairSecret = new Secret(stack, 'AgentNodeKeyPair', {
+      secretName: 'jenkins-agent-node-key-pair',
+      description: 'ssh key for jenkins main node to connect to agent nodes',
+    });
+    Tags.of(keyPairSecret).add('jenkins:credentials:type', 'sshUserPrivateKey');
```
Comment (gaiksaya) [inline lib/compute/agent-node-config.ts:67]: So we need this tag for both secret as well as actual ec2 key pair?

Comment (peterzhuamazon) [reply lib/compute/agent-node-config.ts:67]: We have to add this tag so that jenkins can read this and use it as ssh key, isnt it?

Comment (rishabh6788) [reply lib/compute/agent-node-config.ts:67]: This tag is mandatory for the secret but not for the actual key-pair. 
Don't see any issue with adding the same tag to key-pair as well.

**Pull request merged**